invenio_rdm_records/ext.py

```diff
@@ -75,7 +80,31 @@ def verify_token():
                 # we need to add the LinkNeed here
                 if hasattr(g, "identity"):
                     g.identity.provides.add(LinkNeed(data["id"]))
-
+            else:
+                if not current_app.config.get("RDM_RESOURCE_ACCESS_TOKENS_ENABLED"):
+                    raise RATFeatureDisabledError()
+                # for records: "api/records/<pid_value>/files/<filename>/content"
+                # for drafts: "api/records/<pid_value>/draft/files/<key>/content"
+                filename = (
+                    request.view_args.get("filename")
+                    or request.view_args.get("key")
+                    or None
+                )
+                pid_value = request.view_args.get("pid_value", None)
+
+                if pid_value and filename:
+                    try:
+                        record = RDMDraft.pid.resolve(pid_value, registered_only=False)
```

Comment:
Not proud of this being here and not in the validation method.... but it causes dependency loop in case of importing `RDMDraft` and `RDMRecord` in the `ResourceAccessToken` class